File: internal/models/dtos/user_profile.go
package dtos

type CreateProfileDto struct {
	FirstName string `json:"first_name" validate:"required,max=255"`
	LastName  string `json:"last_name" validate:"required,max=255"`
	Alias     string `json:"alias" validate:"required,max=255"`
	Bio       string `json:"bio" validate:"omitempty,max=255"`
	BirthDate int64  `json:"birth_date" validate:"required,gt=0"`
}

type UpdateProfileDto struct {
	FirstName string `json:"first_name" validate:"required,max=255"`
	LastName  string `json:"last_name" validate:"required,max=255"`
	Alias     string `json:"alias" validate:"required,max=255"`
	Bio       string `json:"bio" validate:"omitempty,max=255"`
	BirthDate int64  `json:"birth_date" validate:"required,gt=0"`
}

File: pages/lib/diva-types/src/user/dtos/user_profile.ts
export type CreateProfileDto = {
  first_name: string;
  last_name: string;
  alias: string;
  bio?: string;
  birth_date: number;
};

export type UpdateProfileDto = {
  first_name: string;
  last_name: string;
  alias: string;
  bio?: string;
  birth_date: number;
};
